plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -272,7 +276,30 @@ private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType ent
                         type)),
                 ColumnType.REGULAR,
                 column.getComment());
-        return TupleDomain.withColumnDomains(ImmutableMap.of(handle, Domain.notNull(handle.getType())));
+
+        ImmutableMap.Builder<HiveColumnHandle, Domain> domains = ImmutableMap.<HiveColumnHandle, Domain>builder()
+                .put(handle, Domain.notNull(handle.getType()));
+        if (entryType == ADD) {
+            partitionConstraint.getDomains().ifPresent(domain -> domain.forEach((key, value) -> domains.put(toPartitionValuesParsedField(handle, key), value)));
+        }
+
+        return TupleDomain.withColumnDomains(domains.buildOrThrow());
+    }
+
+    private static HiveColumnHandle toPartitionValuesParsedField(HiveColumnHandle addColumn, DeltaLakeColumnHandle partitionValueParsedColumn)
+    {
+        return new HiveColumnHandle(
+                addColumn.getBaseColumnName(),
+                addColumn.getBaseHiveColumnIndex(),
+                addColumn.getBaseHiveType(),
+                addColumn.getBaseType(),
+                Optional.of(new HiveColumnProjectionInfo(
+                        ImmutableList.of(0, 0), // hiveColumnIndex; we provide fake value because we always find columns by name
+                        ImmutableList.of("partitionvalues_parsed", partitionValueParsedColumn.getColumnName()),
+                        HiveType.toHiveType(partitionValueParsedColumn.getType()),
```

Comment:
```suggestion
DeltaHiveTypeTranslator.toHiveType(partitionValueParsedColumn.getType())
```

Add a test resource checkpoint containing all the supported partition types - see a similar test in https://github.com/trinodb/trino/pull/19662

I'm thinking that the current code could fail for a `timestamp(3) with time zone` column type.